pbm/client.go

```diff
@@ -94,3 +95,170 @@ func (c *Client) CheckRequirements(ctx context.Context, hubs []types.PbmPlacemen
 
 	return res.Returnval, nil
 }
+
+func (c *Client) GetCompatibleDatastores(ctx context.Context, datastores []vim.ManagedObjectReference, ref *types.PbmServerObjectRef, ids []types.PbmProfileId) ([]vim.ManagedObjectReference, error) {
+	var hubs []types.PbmPlacementHub
+	var req []types.BasePbmPlacementRequirement
+	var compatibleDatastores []vim.ManagedObjectReference
+
+	for _, ds := range datastores {
+		hubs = append(hubs, types.PbmPlacementHub{
+			HubType: ds.Type,
+			HubId:   ds.Value,
+		})
+	}
+
+	for _, id := range ids {
+		req = append(req, &types.PbmPlacementCapabilityProfileRequirement{
+			ProfileId: id,
+		})
+	}
+
+	compatibilityResult, err := c.CheckRequirements(ctx, hubs, ref, req)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, res := range compatibilityResult {
+		if len(res.Error) == 0 {
+			compatibleDatastores = append(compatibleDatastores, vim.ManagedObjectReference{
+				Type:  res.Hub.HubType,
+				Value: res.Hub.HubId,
+			})
+		}
+	}
+
+	return compatibleDatastores, nil
+}
+
+func (c *Client) GetNonCompatibleDatastores(ctx context.Context, datastores []vim.ManagedObjectReference, ref *types.PbmServerObjectRef, ids []types.PbmProfileId) ([]vim.ManagedObjectReference, error) {
+	var hubs []types.PbmPlacementHub
+	var req []types.BasePbmPlacementRequirement
+	var nonCompatibleDatastores []vim.ManagedObjectReference
+
+	for _, ds := range datastores {
+		hubs = append(hubs, types.PbmPlacementHub{
+			HubType: ds.Type,
+			HubId:   ds.Value,
+		})
+	}
+
+	for _, id := range ids {
+		req = append(req, &types.PbmPlacementCapabilityProfileRequirement{
+			ProfileId: id,
+		})
+	}
+
+	compatibilityResult, err := c.CheckRequirements(ctx, hubs, nil, req)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, res := range compatibilityResult {
+		if len(res.Error) > 0 {
+			nonCompatibleDatastores = append(nonCompatibleDatastores, vim.ManagedObjectReference{
+				Type:  res.Hub.HubType,
+				Value: res.Hub.HubId,
+			})
+		}
+	}
+
+	return nonCompatibleDatastores, nil
+}
+
+func (c *Client) CreatePbmProfile(ctx context.Context, capabilityProfileCreateSpec types.PbmCapabilityProfileCreateSpec) (types.PbmProfileId, error) {
+	req := types.PbmCreate{
+		This:       c.ServiceContent.ProfileManager,
+		CreateSpec: capabilityProfileCreateSpec,
+	}
+
+	res, err := methods.PbmCreate(ctx, c, &req)
+	if err != nil {
+		var profileID types.PbmProfileId
```

Comment:
Change the method to return type `*types.PbmProfileId`, then you can use `nil` here and `&res.ReturnVal` below.